libs/infinity_emb/infinity_emb/transformer/embedder/optimum.py

```diff
@@ -25,43 +29,80 @@
         from optimum.onnxruntime import (  # type: ignore[import-untyped]
             ORTModelForFeatureExtraction,
         )
+        from infinity_emb.transformer.utils_optimum import get_onnx_files
 
     except (ImportError, RuntimeError, Exception) as ex:
         CHECK_ONNXRUNTIME.mark_dirty(ex)
 
+
+if CHECK_OPTIMUM_INTEL.is_available:
+    try:
+        from optimum.intel import OVModelForFeatureExtraction  # type: ignore[import-untyped]
+        from infinity_emb.transformer.utils_optimum import get_openvino_files
+
+    except (ImportError, RuntimeError, Exception) as ex:
+        CHECK_OPTIMUM_INTEL.mark_dirty(ex)
+
+
 if CHECK_TRANSFORMERS.is_available:
     from transformers import AutoConfig, AutoTokenizer  # type: ignore[import-untyped]
 
 
 class OptimumEmbedder(BaseEmbedder):
     def __init__(self, *, engine_args: EngineArgs):
-        CHECK_ONNXRUNTIME.mark_required()
         provider = device_to_onnx(engine_args.device)
+        self.provider = provider
+
+        if provider == "OpenVINOExecutionProvider":
+            CHECK_OPTIMUM_INTEL.mark_required()
+            filename = ""
+            try:
+                openvino_file = get_openvino_files(
+                    model_name_or_path=engine_args.model_name_or_path,
+                    revision=engine_args.revision,
+                    use_auth_token=True,
+                )
+                filename = openvino_file.as_posix()
+            except Exception as e:  # show error then let the optimum intel compress on the fly
+                print(str(e))
+
+            self.model = optimize_model(
```

Comment:
@michaelfeil you can just load your model with `OVModelForFeatureExtraction` (if the model wasn't already exported to the OV IR then optimum will load the pytorch model and export it on-the-fly, if already converted it will just load the OV model) without calling `optimize_model` which seems to call `ORTOptimizer` (which should be used for onnx models only). OpenVINO conversion can also be done from an onnx model (https://github.com/huggingface/optimum-intel/blob/6dbc59eb80ba7eee9d347d03f3b737fc54b46e5d/optimum/intel/openvino/modeling_base.py#L347) but usually not recommended and this feature will likely be removed from optimum-intel in the future. Let us know if we can help on this integration!